### DISCUSSION TITLE
`navigationRef` is necessary!

### DISCUSSION THREAD

**Discussion by likeSo** [expo/router#616] [Ideas]
Is page jump behavior only happen inside the JSX tree?

When the network responds to 401 status, how can I call `resetRoot`? (I handled the networking response globally)

For now, I have to use the `EventEmitter`, from the networking file to the `/app/_layout.tsx` file.

Why can't just provide the `navigationRef` from `React Navigation`?

**Comment by NandoMB** [comment]
Something like this, right?

https://reactnavigation.org/docs/navigating-without-navigation-prop

It would be great...

**Comment by wcastand** [comment]
this line in expo-router v2 announcement should fix that no? https://blog.expo.dev/introducing-expo-router-v2-3850fd5c3ca1

```
The new router export can be used as an imperative API to navigate from global stores and providers.
```

**Reply by likeSo** [reply]
I think so, but I cannot find a document about this, any help?

**Reply by wcastand** [reply]
i think it's as simple as doing 

```
import { useRouter } from "expo-router"

useRouter().push("/welcome")
```

**Reply by likeSo** [reply]
Wow... I just checked the source code for `useRouter`, seems like that it's not a hook anymore! Thank you

**Reply by wcastand** [reply]
yeah i guessed that mostly because `zustand` has the same behaviour with their hooks
